appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java

```diff
@@ -311,667 +124,72 @@ private void createBuildMenu() {
-      if (!Strings.isNullOrEmpty(tosUrl)) {
-        html += "<BR/><BR/><a href=\"" + tosUrl +
-            "\" target=\"_blank\">" + MESSAGES.privacyTermsLink() + "</a>";
-      }
-      HTML message = new HTML(html);
+    buildDropDown.removeUnneededSeparators();
 
-      SimplePanel holder = new SimplePanel();
-      Button ok = new Button("Close");
-      ok.addClickListener(new ClickListener() {
-        public void onClick(Widget sender) {
-          db.hide();
-        }
-      });
-      holder.add(ok);
-      DialogBoxContents.add(message);
-      DialogBoxContents.add(holder);
-      db.setWidget(DialogBoxContents);
-      db.show();
+    if (!Ode.getUserAutoloadProject()) {
+      settingsDropDown.setItemHtmlById("AutoloadLastProject", MESSAGES.enableAutoload());
+      settingsDropDown.setCommandById("AutoloadLastProject", new EnableAutoloadAction());
     }
-  }
-
-  private static class AboutCompanionAction implements Command {
-    @Override
-    public void execute() {
-      final DialogBox db = new DialogBox(false, true);
-      db.setText("About The Companion");
-      db.setStyleName("ode-DialogBox");
-      db.setHeight("200px");
-      db.setWidth("400px");
-      db.setGlassEnabled(true);
-      db.setAnimationEnabled(true);
-      db.center();
-
-      String downloadinfo = "";
-      if (!YaVersion.COMPANION_UPDATE_URL1.equals("")) {
-        String url = Window.Location.getProtocol() + "//" + Window.Location.getHost()
-            + YaVersion.COMPANION_UPDATE_URL1;
-        downloadinfo = "<br/>\n<a href=" + url + ">Download URL: " + url + "</a><br/>\n";
-        downloadinfo += BlocklyPanel.getQRCode(url);
-      }
-
-      VerticalPanel DialogBoxContents = new VerticalPanel();
-      HTML message = new HTML(
-          "Companion Version " + BlocklyPanel.getCompVersion() + downloadinfo
-      );
-
-      SimplePanel holder = new SimplePanel();
-      Button ok = new Button("Close");
-      ok.addClickListener(new ClickListener() {
-        public void onClick(Widget sender) {
-          db.hide();
-        }
-      });
-      holder.add(ok);
-      DialogBoxContents.add(message);
-      DialogBoxContents.add(holder);
-      db.setWidget(DialogBoxContents);
-      db.show();
+    if (!Ode.getUserDyslexicFont()) {
+      settingsDropDown.setItemHtmlById("DyslexicFont", MESSAGES.enableOpenDyslexic());
+      settingsDropDown.setCommandById("DyslexicFont", new SetFontDyslexicAction());
     }
-  }
-
-  private static class CompanionUpdateAction implements Command {
-    @Override
-    public void execute() {
-      DesignToolbar.DesignProject currentProject = Ode.getInstance().getDesignToolbar().getCurrentProject();
-      if (currentProject == null) {
-        Window.alert(MESSAGES.companionUpdateMustHaveProject());
-        return;
-      }
-      DesignToolbar.Screen screen = currentProject.screens.get(currentProject.currentScreen);
-      screen.blocksEditor.updateCompanion();
+    if (iamChromebook) {
```

Comment:
We might be able to use the same trick for the Chromebook specific UI as we do for the read only access.